mono/mini/debugger-agent.c

```diff
@@ -1234,11 +1234,20 @@ socket_transport_connect (const char *address)
 	listen_fd = -1;
 
 	if (host) {
+		int hints[] = {
+			MONO_HINT_IPV4 | MONO_HINT_NUMERIC_HOST,
+			MONO_HINT_IPV6 | MONO_HINT_NUMERIC_HOST,
+			MONO_HINT_UNSPECIFIED
+		};
 
 		mono_network_init ();
 
-		/* Obtain address(es) matching host/port */
-		s = mono_get_address_info (host, port, MONO_HINT_UNSPECIFIED, &result);
+		for (int i = 0; i < sizeof(hints) / sizeof(int); i++) {
```

Comment:
Shouldn't we check that the address is actually numeric before doing this ?

Reply:
That's what the loop does. The flags we specify will make `getaddrinfo` use `inet_pton` and this way of doing it is, for us, less cumbersome and more elegant - we don't have to introduce a special case for IP-as-host-name (an extra `sockaddr` variable + a flag). In most cases we'll get IP address (IDEs always use the IP address) so the first call will succeed and there's no time lost.

Reply:
`getaddrinfo ()` will do that internally.